LibAddonMenu-2.0/LibAddonMenu-2.0/controls/description.lua

```diff
@@ -3,16 +3,38 @@
     text = "My description text to display.", -- or string id or function returning a string
     title = "My Title", -- or string id or function returning a string (optional)
     width = "full", --or "half" (optional)
+    disabledDesc = function() return db.someBooleanSetting end, --or boolean (optional)
     reference = "MyAddonDescription" -- unique global reference to control (optional)
 } ]]
 
 
-local widgetVersion = 8
+local widgetVersion = 9
 local LAM = LibStub("LibAddonMenu-2.0")
 if not LAM:RegisterWidget("description", widgetVersion) then return end
 
 local wm = WINDOW_MANAGER
 
+local function GetResultFromValueOrFunction(value)
+    if type(value) == "function" then
+        return value()
+    else
+        return value
+    end
+end
+
+local function GetColorForState(disabled)

```

Comment:
use `local GetColorForState = LAM.util.GetColorForState` instead (you'll have to rebase onto the new commits).